tf2_pykdl_ros/src/tf2_pykdl_ros/tf2_pykdl_ros.py

```diff
@@ -0,0 +1,185 @@
+from typing import Union
+from geometry_msgs.msg import PointStamped, PoseStamped, TransformStamped
+import PyKDL as kdl
+from pykdl_ros import VectorStamped, FrameStamped
+import tf2_ros
+
+
+def transform_to_kdl(t: TransformStamped) -> kdl.Frame:
+    """
+    Convert a geometry_msgs Transform message to a PyKDL Frame.
+
+    :param t: The Transform message to convert.
+    :type t: geometry_msgs.msg.TransformStamped
+    :return: The converted PyKDL frame.
+    :rtype: PyKDL.Frame
+    """
+
+    return kdl.Frame(
+        kdl.Rotation.Quaternion(
+            t.transform.rotation.x, t.transform.rotation.y, t.transform.rotation.z, t.transform.rotation.w
+        ),
+        kdl.Vector(t.transform.translation.x, t.transform.translation.y, t.transform.translation.z),
+    )
+
+
+def to_msg_vector(vector: VectorStamped) -> PointStamped:
+    """
+    Convert a VectorStamped to a geometry_msgs PointStamped message.
+
+    :param vector: The vector to convert.
+    :type vector: pykdl_ros.VectorStamped
+    :return: The converted vector/point.
+    :rtype: geometry_msgs.msg.PointStamped
+    """
+    msg = PointStamped()
+    msg.header = vector.header
+    msg.point.x = vector.vector[0]
+    msg.point.y = vector.vector[1]
+    msg.point.z = vector.vector[2]
+    return msg
+
+
+tf2_ros.ConvertRegistration().add_convert((VectorStamped, PointStamped), to_msg_vector)
+tf2_ros.ConvertRegistration().add_to_msg(VectorStamped, to_msg_vector)
+
+
+def from_msg_vector(msg: Union[PointStamped, PoseStamped]) -> VectorStamped:
+    """
+    Convert a PointStamped/PoseStamped message to a stamped VectorStamped.
```

Comment:
Why would we want to convert a PoseStamped msg to a VectorStamped object? Then you lose data? Better to have two functions

Reply:
True, you lose data. But I think the user should be aware of that herself/himself. Because of the registration in tf2_ros, it is not possible to have two functions. As we can only register one from_msg function per class type.